coding-concepts.js

```diff
@@ -6,49 +6,106 @@
+
+//      the correct answer was 5 for the total number of strings in the projected code.
+
+//
+
 // --------------------1) What will this log?
 
-const cohort = "LEARN 2023"
+// const cohort = "LEARN 2023"
 // console.log(cohort.length)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: 10
+// b) Verify and explain: 
+
+//      My guess:
+
+//      I say its going to be 10. The .length function call count in an Index format and it starts at 0 instead on 1. 
+//      There is 11 characters in the string so it will the character from 0 to 10. So the answer is 10.
 
 // --------------------2) What will this log?
 
-const greeting = "Hello World!"
+// const greeting = "Hello World!"
 // console.log(greeting[4])
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: o
+// b) Verify and explain: 
+
+        // My guess:
+
+        // The console.log function call has the variable in the argument with the number 4 in the brackets. The variable
+        // reference the string "Hello World". The 4 will count the string in an Index format from 0 to 4 so the answer should be "o"
 
 // --------------------3) What will this log?
 
-const languages = ["JavaScript", "Ruby", "Python", "C++"]
-const index = 1
+// const languages = ["JavaScript", "Ruby", "Python", "C++"]
+// const index = 1
 // console.log(languages[index])
 
-// a) Your answer:
+// a) Your answer: Ruby
 // b) Verify and explain:
 
+//  My guess: 
+
+//  We declare the variable "languages" equal to the array.
+
+//  In the argument, we have the index variable that is equal to the value (1).
+
+//  The array is counting the string from an index format from 0 to 1. 
+
+//  The answer should be "Ruby"
+
 // --------------------4) What will this log?
 
-const weekendDays = ["saturday", "sunday"]
+// const weekendDays = ["saturday", "sunday"]
 // console.log(weekendDays.toUpperCase())
 
-// a) Your answer:
+// a) Your answer: SATURDAY, SUNDAY
 // b) Verify and explain:
 
+//  My guess:
+
+//  I think that it will display in all caps. the variable in the fuction call console.log() had both strings in the array
+//  with the .toUpperCase() built-in method that capitalize the elements in the array.
+
 // --------------------5) What will this log?
 
 const dataTypes = ["number", "string", "Boolean", "undefined"]
-// console.log(typeof dataTypes.length)
+console.log(typeof dataTypes.length)
 
-// a) Your answer:
+// a) Your answer: 3
 // b) Verify and explain:
+
+//  I dont know what "typeof" is suppose to do i never seen it or i don't recall. I didnt see it in the notes but I will assume that,
+//  that is what it's going to do.
+
+//  Incorrect
+
+//  The correct answer was "number"
+
+//  I did some research and found out that type of is an operator that find out the data typ. They're strings but I dont know what is
+//  making the answer come up with number.
```

Comment:
`typeof` is looking at `dataTypes.length`. Since dataTypes.length gives us the length of the array, it will give us `4`. Then, `typeof` looks at `4` and returns its data type, which is number.